src/test/java/de/tum/in/www1/artemis/architecture/ArchitectureTest.java

```diff
@@ -202,6 +202,18 @@ void testJSONImplementations() {
                 .check(allClasses);
     }
 
+    @Test
+    void testDTOImplementations() {
+        var dtoRecordRule = classes().that().haveSimpleNameEndingWith("DTO").should().beRecords().andShould().beAnnotatedWith(JsonInclude.class)
+                // .andShould(useJsonIncludeNonEmpty())
+                .because("All DTOs should be records and annotated with @JsonInclude(JsonInclude.Include.NON_EMPTY)");
```

Comment:
Why was this line commented out? If we want to re-add these checks we should at least add a TODO. if it should stay removed we should adjust the failure message accordingly.
```suggestion
        var dtoRecordRule = classes().that().haveSimpleNameEndingWith("DTO").should().beRecords().andShould().beAnnotatedWith(JsonInclude.class)
                // .andShould(useJsonIncludeNonEmpty())
                .because("All DTOs should be records and annotated with @JsonInclude");
```

Reply:
it was not working properly and there is already another check that all JsonInclude use the non empty version